client/notices.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/url"
+	"time"
+)
+
+type NotifyOptions struct {
+	// Key is the custom notice's key. Must be in "domain.com/key" format.
+	Key string
```

Comment:
This is hiding the fact we have a Type here, and making it an internal option, but then when the notice comes back we suddenly have the Type detail available.  I think we should make these details known and used here, even if for now only a CustomNotice type can be emitted.

Reply:
Yeah, fair enough -- I've added `Type` (the server ensures it's set and equal to "custom").